lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -53,6 +53,24 @@ class LoadBalancer
 end
 end
 
+require 'chef/provider/load_balancer'
+class Chef
+class Provider
+  class LoadBalancer
+    # We override this so we can specify a machine name as `i-123456`
+    # This is totally a hack until we move away from base resources
+    def get_machine_spec!(machine_name)
+      if machine_name =~ /^i-[a-f0-9]{8}$/
```

Comment:
Is this the same across all regions? ISTR there being different lengths, if nothing else.